code/contracts/ONEWallet.sol

```diff
@@ -105,13 +109,15 @@ contract ONEWallet {
             bytes32(bytes4(indexWithNonce)),
             eotp
         );
-        bytes32 commitHash = keccak256(bytes.concat(packed));
+        bytes32 commitHash = keccak256(bytes.concat(packed)); // IH: why do you need bytes.concat again?
         _revealPreCheck(commitHash, indexWithNonce);
         _completeReveal(commitHash);
+        // IH: until here, the code is duplicate with revealTransfer(). Can you put it into some helper function?
```

Comment:
There isn't a way to get a list of arguments, otherwise we could use a common modifier